frontend/src/metabase/visualizations/echarts/cartesian/model/series.ts

```diff
@@ -162,53 +162,56 @@ export const getCardSeriesModels = (
 
   // Charts without breakout have one series per selected metric column.
   if (!hasBreakout) {
-    return columns.metrics.map((metric) => {
-      const vizSettingsKey = getSeriesVizSettingsKey(
-        metric.column,
-        hasMultipleCards,
-        isFirstCard,
-        columns.metrics.length,
-        null,
-        card.name,
-      );
-      const legacySeriesSettingsObjectKey =
-        createLegacySeriesObjectKey(vizSettingsKey);
-
-      const customName = settings[SERIES_SETTING_KEY]?.[vizSettingsKey]?.title;
-      const tooltipName = customName ?? metric.column.display_name;
-      const name =
-        customName ??
-        getDefaultSeriesName(
-          metric.column.display_name,
+    return columns.metrics
+      .filter((m) => !!m.column)
```

Comment:
Only true change in this file is here. Before data is fully loaded, columns might be undefined here.

Reply:
It would be great to ensure we don't even get to render this cartesian chart component if the data is incomplete, wdyt?